### ISSUE TITLE
Many thanks for this!

### ISSUE THREAD

**Issue by ramapcsx2** [filipalac/240pTestSuite-PS1#1]
This port is already helping nicely with my other work. It's great to have the PSX (and PS2!) available for image quality tests.

To make this issue worth it, here's a couple notes:
- On the PS2, button inputs are somewhat delayed, but the suite does work.
- The scroll test has an odd flicker on some of the background details. You probably noticed.
- It'd be great to have all the typical horizontal resolutions, at least in some of the tests. It would be most useful in the colorbleed and checkerboard patterns (to me, at least).

Thanks again! :)

**Comment by filipalac** [comment]
Firstly sorry for my very late reply and thanks for your feedback.

- I tested suite on slim SCPH-77004 and sadly (or happily :D) didn't noticed any delay.
- Thanks, i would probably never notice that :).

I just committed new version with fixed sonic background and colorbleed, checkerboard and horizontal lines (and vertical) patterns with extra support of 384 and 640 resolutions.

Thanks again for your feedback and suggestion.

**Comment by ramapcsx2** [comment]
Nice, I'll check it out!

Looking at those screens a ton all day (other project :p), the background in the Sonic stage is some pixels in the trees and water that color cycle wrong somehow. Maybe it's already fixed.

The delay with the pad inputs is somehow the PS2 DualShock controller (which I use on PSX and PS2).
It works great on PSX but on the PS2, it seems to report too many values at once for the button pressed part of the code. (I haven't looked at that, sorry.)
Since the update with 256 pix wide support, all the face buttons don't work at all, unless the DualShock is in analog mode.
If it is, the face buttons only work sporadically. I have to keep the buttons pressed and they will randomly register.

Thanks again! :)

**Comment by filipalac** [comment]
I tried to fix the problem but sadly no luck, this is what I tried:
- Poll pad every second frame.
- Example programs provided by PSXSDK.
- ShendoXT padtest program.
- Airport game by XaviDCR90 with his own fork of PSXSDK.

When printing parameters of psx_pad_state status, .id, and .type work as expected but with .buttons and .extra behavior is influenced by type of controller (scph 10010H and M). 

- On first controller moving dpad up or down is pressing X or O. Moving up and down with right analog is controlling x-axis  and left analog is controlling y-axis. All buttons are changing .extra value. 

- On second controller dpad is responding (only if analog is turned off), but  very inconsistently (output value is "blinking"), when analog is turned on, it move all the way to right and ignoring input.

Honestly i don't know why it isn't working and especially have no idea why every controller is retarded in his own way. Maybe as a workaround I can try to figure how to use bios joypad function and hope for best. Or find true cause of bug and fix it :).

I just want to say don't expect bugfix soon, school starting 8.1 and I have lot of homework
and stuff to do :(. Oh and it works fine with psx controller and third party ps2 controller.

**Comment by ramapcsx2** [comment]
The new features are great and the Sonic stage glitch is also gone. Nice work! :)
Don't worry too much about the DualShock 2. It seems it won't be an easy fix and it still kind of works.

**Comment by James-F2** [comment]
@filipalac
Bless you dear sir for making this happen.
This is just perfect for adjusting the OSSC for PS1 and PS2.

**Comment by ramapcsx2** [comment]
I've got another request! ;p
Could you send a CDSTOP command once the program has loaded? I noticed it loads everything to RAM on bootup anyway.
Right now the disk keeps spinning in 2x mode, with the mechanics working hard to keep tracking up.
Since this software may run for hours at a time (it does here :) ), that will eventually break these old drives.

**Comment by James-F2** [comment]
> Could you send a CDSTOP command once the program has loaded?

+1

**Comment by filipalac** [comment]
Feature added.

**Comment by ramapcsx2** [comment]
Cheers! ;)

**Comment by James-F2** [comment]
Thanks!

Some idiot removed the Playstation 240p suite link from junkerhq wiki page.
Can you please add it back?
http://junkerhq.net/xrgb/index.php?title=240p_test_suite

**Comment by ramapcsx2** [comment]
That's odd that it got removed. Maybe he's just updating files? The port is pretty much perfect and I assume there's high demand for this version as well.

@filipalac 
I just burned the latest version and I'm happy to report that the CD stop function works as expected. That should save a few drives :)
Did you develop this in a VM by chance? Last I tried, I had trouble setting up and using the SDK.
I want to maybe extend this suite with some customizations that help diagnose hardware issues.
I figure that with the 240p test suite as a base, this could then become an all-in-one PSX test CD.
Features I'd like to add:
- CD ROM test commands that aid with laser repairs (disable tracking, run autofocus routine, etc)
- PSX CPU / RAM 100% load feature, to spot failing power supplies and capacitors (this would show up as noise on the display)

**Comment by ArtemioUrbina** [comment]
The changes to the wiki were reverted by a backup restore from the database, and non intentional. 

It is back in shape now, sorry about that.

**Comment by filipalac** [comment]
@ramapcsx2 
No, I built it from the source and I'm using it on Arch Linux, feel free to add new features.

**Comment by ArtemioUrbina** [comment]
@ranapcsx2 that'd be great.  We've extended the suite that same way in the past.

**Comment by carlosrobayo23** [comment]
Sorry, newb question. I found the bin/cue files, But I can not open it in WinISO or any image explorer. Do I need to do something aditional to burn the CD?

**Comment by filipalac** [comment]
On Windows, I know ImgBurn should work fine. Just feed it .cue and it will pick up .bin itself.

**Comment by nonijod** [comment]
Tried it on my 30004R PS2, however the application is all glitchy and laggy. Vid for illustration - https://youtu.be/McQ1SFO3aRc

**Comment by filipalac** [comment]
Thanks for bug report.
However I probably wont fix (more like workaround) this bug. I can't reproduce it on my slim, and Sony really screwed up their ps1 emulation. Like if I try to workaround it with some nasty hacks, who knows if some other bug(s) won't show up. Especially when different ps2 revisions emulate ps1 little bit different.

**Comment by James-F2** [comment]
Works perfectly on my PSOne.
Yes the controller navigation is a little laggy on my PS2 Slim, but no graphic glitches.

**Comment by filipalac** [comment]
What controller were you testing with ps2? By little laggy you mean something like 1-2 frames?

My testings showed original ps2 controllers to be REALLY broken. 3rd party ps2 controller and ps1 dualshock worked flawlessy.
And to be honest, I experienced little graphic glitches in some help texts, not all words were showing properly.

**Comment by carlosrobayo23** [comment]
Did anyone try with PSP/PSPGO by creating the eboot of this bin+cue?
